### PR TITLE
#3623 [Manager] Wrap the fileApi, refactor the FileContent

### PR DESCRIPTION
**Pull request by eechih**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3623 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Workspace/WorkspaceQuick.js [hunk 1]
```diff
@@ -142,9 +144,11 @@ const WorkspaceQuick = props => {
   };
 
   const onDrop = async (file, values) => {
+    const parentKey = { group: WORKSPACE, name: values.workspaceName };
     const result = await fileApi.create({
-      group: values.workspaceName,
+      group: hashWith(WORKSPACE, values.workspaceName),
       file: file[0],
+      tags: { parentKey },
     });
```
Comment (eechih) [inline ohara-manager/client/src/components/Workspace/WorkspaceQuick.js]: 在建立新 Workspace 過程中，此時 AppContext 無法取得即將建立 workspace 的 name，因此無法使用 ApiContext 的 fileApi 上傳 file，因為 file 的 group，必須仰賴 workspaceName 計算而得。

@saivirtue 與其為了「建立 workspace」打開後門，而破壞 AppContext 的封裝，我個人比較頃向直接跳過 ApiContext，直接使用 api。

File: ohara-manager/client/src/context/api/fileApi.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      const params = { name, group };
+      const res = await fileApi.get(params);
+      if (!isEmpty(res.errors)) {
+        throw new Error(res.title);
+      }
+      return res.data;
+    },
+    create: async file => {
+      try {
+        const params = { file, group, tags: { parentKey } };
```
Comment (eechih) [inline ohara-manager/client/src/context/api/fileApi.js]: parentKey 僅是用來記錄與上層物件的關係，比較能用肉眼看。

File: ohara-manager/client/src/utils/sha.js [hunk 3]
```diff
@@ -28,4 +29,11 @@ export const hash = (input = '', options = defaultHashOptions) =>
     .digest(options.encode)
     .substring(0, options.maxLength + 1);
 
+export const hashBy = (object, paths = []) => {
+  if (isEmpty(paths)) throw new Error('The paths must not be an empty.');
+  return hash(join(map(paths, path => get(object, path))));
```
Comment (konekoya) [inline ohara-manager/client/src/utils/sha.js]: Good one, FP style :)

Comment (vitojeng) [reply ohara-manager/client/src/utils/sha.js]: 👍

**Pull request merged**